thentos-core/src/Thentos.hs

```diff
@@ -72,6 +73,23 @@ makeMain commandSwitch =
         mFeConfig :: Maybe HttpConfig
         mFeConfig = Tagged <$> config >>. (Proxy :: Proxy '["frontend"])
 
+        csrfSecret :: Maybe ST
+        csrfSecret = config >>. (Proxy :: Proxy '["csrf_secret"])
+
+    when (isJust mFeConfig && not (validFormatCsrfSecretField csrfSecret)) $ do
+        CsrfSecret freshCsrfSecret <- genCsrfSecret
+        throwIO . ErrorCall . unlines $
+          [ "The field csrf_secret is currently " <>
+            if isNothing csrfSecret then "missing." else "invalid."
+          , "I just securely generated such a secret for you:"
+          , "  " <> show freshCsrfSecret
+          , ""
+          , "You can either use the field csrf_secret field from the configuration, or the"
+          , "CSRF_SECRET environement variable, or the command line flag --CSRF_SECRET."
```

Comment:
Reading configifier sources, this may even be thentos_CSRF_SECRET or something similar, and I actually didn't see code switching it into upper case. So a (manual) test would be very welcome.